alacritty_terminal/src/term/mod.rs

```diff
@@ -2089,6 +2102,24 @@ impl<T: EventListener> ansi::Handler for Term<T> {
         trace!("Setting cursor style {:?}", style);
         self.cursor_style = style;
     }
+
+    #[inline]
+    fn push_title(&mut self) {
+        let title = self.title.to_owned();
+        trace!("Pushing '{}' onto title stack", title);
+        if self.title_stack.len() >= TITLE_STACK_MAX_DEPTH {
+            self.title_stack.remove(0);
+        }
+        self.title_stack.push(title);
+    }
```

Comment:
This function would benefit from a bit of reordering and empty lines to improve clarity.

First of all we can just put the `trace` right at the top with a newline separating it from everything else and printing `self.title`, we don't need an owned string for that.

Then we don't need to actually define the title as variable at all, it's only used in one place and we can inline it instead.

Instead of using `to_owned` we should also use `clone` instead. This makes it clear that we are cloning an existing string rather than converting a `&str` to a `String`.